pecos/xmc/xlinear/model.py

```diff
@@ -299,6 +299,53 @@ def split_model_at_depth(self, given_depth, reindex=False):
             )
         return split_model
 
+    @classmethod
+    def reconstruct_model(cls, parent_model, child_models, Y_ids_of_subtrees):
```

Comment:
We should add unit test for this function. In particular, `reconstruct_model` should be able to restore the model splited via `split_model_at_depth`.